src/network/networkd-link.c

```diff
@@ -3453,6 +3487,17 @@ static int link_carrier_lost(Link *link) {
         if (r < 0)
                 return r;
 
+        /* It may be that the administrator is trying to manage the interface
+         * outside systemd-networkd, but set ActivationMode=on. Try to bring
+         * the link up again in case it's not really a carrier change.
+         */
+        if (link_should_activate(link) && !link_is_manual(link)) {
```

Comment:
It's not current behavior, but AIUI it's the **wanted** behavior, consistent with networkd managing everything. My understand of the systemd-networkd philosophy (from earlier contributions) is that by default, it manages things, is authoritative (well, aside from the kernel) as to the network configuration, and doesn't care what you do outside. In other words, running 'ip link set dev <> down' is _not_ supported. If carrier is lost because the physical carrier is lost, then bringing the link back up won't hurt, it still won't have carrier. If the carrier is lost because the user ran a command to circumvent systemd, then systemd can stay in charge, and bring it back up.

What I want is for 'ip link set dev <> down' and 'ip link set dev <> up' to be honoured and officially supported; that's what `ActivationMode=manual` achieves. In that mode, systemd-networkd manages the network normally, will try to bring interfaces up by default (you configured them, therefore you likely want them up as much as possible), but it won't revert outside changes.

If I'm wrong about what the default is meant to be, then we can still change the default.